src/main/java/com/example/quizapp/quiz/QuizResultController.java

```diff
@@ -0,0 +1,125 @@
+package com.example.quizapp.quiz;
+
+import com.example.quizapp.NavigationStack;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.geometry.Pos;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+import javafx.scene.text.Font;
+import org.javatuples.Triplet;
+
+import java.io.IOException;
+
+/**
+ * Represents the results of a quiz
+ */
+public class QuizResultController extends AnchorPane {
+
+    @FXML Label resultLabel;
+    @FXML Label percentLabel;
+    @FXML ScrollPane resultPane;
+    @FXML VBox container;
+
+    private final Triplet<String, String, String>[] takenQuiz;
+    NavigationStack navigationStack = NavigationStack.getInstance();
+
+    /**
+     *
+     * Loads the QuizSummery fxml file and initialize values for the current object with the specified arguments
+     * @param takenQuiz the quiz that was taken
+     * @param points the total points obtained
+     * @param fullScore the total score of the entire quiz
+     */
+    public QuizResultController(Triplet<String, String, String>[] takenQuiz, int points, int fullScore){
+
+        FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("/fxml/QuizSummery.fxml"));
+        fxmlLoader.setRoot(this);
+        fxmlLoader.setController(this);
+
+        try {
+            fxmlLoader.load();
+        } catch (IOException exception) {
+            throw new RuntimeException(exception);
+        }
+
+        this.takenQuiz = takenQuiz;
+
+        resultPane.setPannable(true);
+        resultPane.setFitToWidth(true);
+        totalPoints(points, fullScore);
+        allQuestions();
+    }
+
+    private void totalPoints(int score, int overAll){
+        resultLabel.setText("Your total points for this quiz is: " + score + "/" + overAll);
+
+        int grade = (score*100) / overAll;
+        percentLabel.setText("Overall: " + grade + "%");
+    }
+
+    private void allQuestions(){
+        int n = takenQuiz.length;
+
+        for (int i = 0; i < n; i++){
```

Comment:
kanske tydligare att ha for(int i = 0; i < takenQuiz.length; i++) eller ett mer beskrivande namn på n :)